packages/wallet/frontend/src/pages/_app.tsx

```diff
@@ -13,6 +18,41 @@ const titilium = Titillium_Web({
 
 export default function App({ Component, pageProps }: AppPropsWithLayout) {
   const getLayout = Component.getLayout ?? ((page) => page)
+  const { toast } = useToast()
+
+  useEffect(() => {
+    let socket: Socket | null = null
+    // Connect to the Socket.IO server
+    socket = io(process.env.NEXT_PUBLIC_BACKEND_URL ?? '', {
+      withCredentials: true
+    })
+
+    // Event listeners
+    socket?.on('connect', () => {
+      // console.log('Connected to server')
+    })
+
+    socket?.on('MONEY_RECEIVED', (data) => {
+      toast({
```

Comment:
what happens if you are on a page in which you see the balance somewhere?
That should be extracted as some sort of global state, and this should update it as well.

Reply:
Just updated with the code to automatically update balance